File: Python/SQLAlchemy_ORM/bll/services/territory_service.py
from dal.repositories.territory_repository import TerritoryRepository
from dal.data.database import SessionLocal
from dal.entities.territory import Territory

class TerritoryService:
    def __init__(self):
        self.db = SessionLocal()
        self.repo = TerritoryRepository(self.db)

    def get_by_id(self, id: str):
        entity = self.repo.get_by_id(id)
        if not entity:
            raise ValueError(f"Entity {id} not found")
        return entity

    def get_all(self):
        return self.repo.get_all()

    def create(self, entity: Territory) -> Territory:
        return self.repo.create(entity)

    def update(self, id: str, entity: Territory) -> Territory:
        entity = self.repo.update(id, entity)
        if not entity:
            raise ValueError(f"Entity {id} not found for update")
        return entity

    def delete(self, id: int):
        entity = self.repo.delete(id)
        if not entity:
            raise ValueError(f"Entity {id} not found for deletion")
        return entity

File: Node/Sequelize_ORM/src/bll/services/territoryService.js
const TerritoryRepository = require('../../dal/repositories/territoryRepository');

class TerritoryService {
  async getAll() {
    return await TerritoryRepository.getAll();
  }

  async getById(id) {
    return await TerritoryRepository.getById(id);
  }

  async create(data) {
    return await TerritoryRepository.create(data);
  }

  async update(id, data) {
    return await TerritoryRepository.update(id, data);
  }

  async delete(id) {
    return await TerritoryRepository.delete(id);
  }
}

module.exports = new TerritoryService(); 